testsuite/MDAnalysisTests/coordinates/test_xdr.py

```diff
@@ -597,116 +603,197 @@ def test_triclinic_box():
+    # This base class assumes same lengths and dt for XTC and TRR test cases!
+    filename = None
+    ref_unitcell = np.array([80.017, 80.017, 80.017, 60., 60., 90.],
+                            dtype=np.float32)
+    # computed with Gromacs: 362.26999999999998 nm**3 * 1000 A**3/nm**3
+    ref_volume = 362270.0
+    ref_offsets = None
+    _reader = None
+
+    def setUp(self):
+        # since offsets are automatically generated in the same directory
+        # as the trajectory, we do everything from a temporary directory
+        self.tmpdir = tempdir.TempDir()
+        shutil.copy(self.filename, self.tmpdir.name)
+
+        self.traj = os.path.join(self.tmpdir.name,
+                                 os.path.basename(self.filename))
+
+        self.trajectory = self._reader(self.traj)
+        self.prec = 3
+        self.ts = self.trajectory.ts
+
+    def tearDown(self):
+        del self.tmpdir
+        del self.trajectory
+
+    @dec.slow
+    def test_offsets(self):
+        self.trajectory._read_offsets(store=True)
+        assert_array_almost_equal(self.trajectory._xdr.offsets,
+                                  self.ref_offsets,
+                                  err_msg="wrong frame offsets")
+
+        outfile_offsets = XDR.offsets_filename(self.traj)
+        with open(outfile_offsets) as f:
+            saved_offsets = {k: v for k, v in np.load(f).iteritems()}
+
+        assert_array_almost_equal(self.trajectory._xdr.offsets,
+                                  saved_offsets['offsets'],
+                                  err_msg="error saving frame offsets")
+        assert_array_almost_equal(self.ref_offsets, saved_offsets['offsets'],
+                                  err_msg="saved frame offsets don't match "
+                                  "the known ones")
+
+        self.trajectory._load_offsets()
+        assert_array_almost_equal(self.trajectory._xdr.offsets,
+                                  self.ref_offsets,
+                                  err_msg="error loading frame offsets")
+        assert_equal(saved_offsets['ctime'], os.path.getctime(self.traj))
+        assert_equal(saved_offsets['size'], os.path.getsize(self.traj))
+
+    def test_reload_offsets(self):
+        self._reader(self.traj, refresh_offsets=True)
+
+    # TODO: actually tests mismatchs, this only ensures the code-path is run
+    @dec.slow
+    def test_persistent_offsets_size_mismatch(self):
+        # check that stored offsets are not loaded when trajectory
+        # size differs from stored size
+        fname = XDR.offsets_filename(self.traj)
+        with open(fname) as f:
+            saved_offsets = {k: v for k, v in np.load(f).iteritems()}
+        saved_offsets['size'] += 1
+        with open(fname, 'w') as f:
+            np.savez(f, **saved_offsets)
+        self._reader(self.traj)
+
+    # TODO: actually tests mismatchs, this only ensures the code-path is run
+    @dec.slow
+    def test_persistent_offsets_ctime_mismatch(self):
```

Comment:
I currently can't test this. For some reason a warning is only issued once for me during a python session. When I trigger the same warning message again I don't see it appear. I have no idea why though.